File: workoutFront/src/apis/response/response.dto.ts
//tsconfig에서baseurl을 설정해 주었기 때문에 아래와 같은 상대경로가 아닌
//절대경로("types")로 경로를 설정할 수 있음 

// import ResponseCode from "../../types/response.code.enum";  
import {ResponseCode} from "../../types/enum";

export default interface ResponseDto{
    code: ResponseCode;
    message: string;  
}

File: workoutbackend/src/main/java/com/flexy/workoutbackend/dto/response/ResponseDto.java
package com.flexy.workoutbackend.dto.response;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import com.flexy.workoutbackend.common.ResponseCode;
import com.flexy.workoutbackend.common.ResponseMessage;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
// 나중에 DTO를 하나하나씩 만들때 ResponseDto를 상속받아 확장해서 만들것입니다.
public class ResponseDto {
    private String code;
    private String message;

    // 모든 response들이 반환하는 타입인 ResponseEntity
    public static ResponseEntity<ResponseDto> databaseError() {
        System.out.println("DB에러");
        ResponseDto responseBody = new ResponseDto(ResponseCode.DATABASE_ERROR, ResponseMessage.DATABASE_ERROR);
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseBody);
    }

    public static ResponseEntity<ResponseDto> validationFailed() {
        ResponseDto responseBody = new ResponseDto(ResponseCode.VALIDATION_FAILED, ResponseMessage.VALIDATION_FAILED);
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(responseBody);
    }

}
